src/s3fs.cpp

```diff
@@ -1166,9 +1166,12 @@ static int s3fs_unlink(const char* _path)
     }
     S3fsCurl s3fscurl;
     result = s3fscurl.DeleteRequest(path);
-    StatCache::getStatCacheData()->DelStat(path);
-    StatCache::getStatCacheData()->DelSymlink(path);
-    FdManager::DeleteCacheFile(path);
+    // delete stat cache when it's not downloading
+    if(fileReading.find(path) == fileReading.end()){
```

Comment:
When are cache files (s3fs' own) deleted?
It seems that there is no timing to delete unless it is executed in this part.